File: src/test/java/de/schramm/royalbash/tdd/core/creature/CreatureUtilTest.java
package de.schramm.royalbash.tdd.core.creature;

import de.schramm.royalbash.tdd.core.Context;
import de.schramm.royalbash.tdd.core.Game;
import de.schramm.royalbash.tdd.core.Player;
import de.schramm.royalbash.tdd.core.Spot;
import lombok.val;
import org.junit.Test;

import java.util.stream.Collectors;

import static org.assertj.core.api.Assertions.assertThat;

public class CreatureUtilTest {

    @Test
    public void should_put_creature_to_player_board() {

        // Given
        val card = NoOpCreature.builder()
                .build();
        val spot = Spot.builder().build();
        val player1 = Player.builder()
                .handcard(card)
                .spot(spot)
                .build();
        val player2 = Player.builder().build();
        val game = Game.builder()
                .player1(player1)
                .player2(player2)
                .build();
        val context = Context.builder()
                .game(game)
                .owner(player1)
                .targetSpot(spot)
                .build();

        // When
        val updatedGame = CreatureUtil.spawnCreature(card, context);

        // Then
        val cardInSpot = updatedGame.getPlayer1().getSpots().collect(Collectors.toList()).get(0).getCreature();
        assertThat(cardInSpot.isPresent()).isTrue();
        assertThat(cardInSpot.orElse(NoOpCreature.builder().build())).isEqualTo(card);
    }
}

File: backend/src/test/kotlin/de/schramm/royalbash/domain/card/creature/CreatureUtilTest.kt
package de.schramm.royalbash.domain.card.creature

import de.schramm.royalbash.domain.*
import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test

class CreatureUtilTest {

    @Test
    fun should_put_creature_to_player_board() {

        // Given
        val card = CreatureMock(id = "Id 1", instanceId = "InstanceId 1")
        val targetSpot = Spot(id = "spot")
        val owner = Player("Id 2", handcards = listOf(card), spots = listOf(targetSpot))
        val player2 = Player("Id 3")
        val game = Game("Id 4", player1 = owner, player2 = player2)
        val context = Context(game, owner, targetSpot = targetSpot)

        // When
        val updatedGame = card.spawnCreature(context)

        // Then
        val cardInSpot = updatedGame
                .player1
                .spots
                .first()
                .creature
        Assertions.assertThat(cardInSpot)
                .isNotNull
                .isEqualTo(card)
    }
}
